app/src/main/java/com/asfoundation/wallet/advertise/WalletPoAService.java

```diff
@@ -493,6 +493,13 @@ public void handlePoaCompletedToSendEvent() {
     if (completedEventDisposable == null || completedEventDisposable.isDisposed()) {
```

Comment:
Can't we handle both scenarios in the same place?
Also, can we extract the logic to a private method so the main method continues readable?

Reply:
this applies for the hole method